src/ProvidedTypes.fs

```diff
@@ -14365,8 +14368,11 @@ namespace ProviderImplementation.ProvidedTypes
 
                 match methodBaseT with
                 | :? ProvidedMethod as mT when (match methodBaseT.DeclaringType with :? ProvidedTypeDefinition as pt -> pt.IsErased | _ -> true) ->
-                    let exprT = mT.GetInvokeCode(Array.toList parametersT)
-                    check exprT
+                    match mT.GetInvokeCode with
+                    | Some invokeCode ->
+                        let exprT = invokeCode(Array.toList parametersT)
+                        check exprT
+                    | None -> <@@ () @@>

```

Comment:
We should raise an exception at this point if a body has not been provided where one will be needed (i.e. when the method is not abstract) - indeed earlier, in the ProvidedMethod constructor.  Likewise we should raise an exception if a body has been provided for a method that can't have one (an interface instance member)